<launch>
<group ns="ground_detector">
    <param name="base_frame" type="string" value="lidar_os2/os_lidar"/>

    <!-- ground detector params -->
    <param name="lidar_topic" type="string" value="/lidar_os2/points"/>
    <param name="filter_topic" type="string" value="pcl_filtered"/>
    <param name="ground_projection" type="string" value=""/>
    <param name="z_offset" type="double" value="-0.08"/>
    <param name="tolerance" type="double" value="0.3"/>
    <param name="range" type="double" value="10.0"/>
    <param name="z_limit" type="double" value="2.7"/>
    <param name="resolution" type="double" value="0.5"/>
    <param name="publish_ground" type="bool" value="true"/>
    <param name="output_topic" type="string" value="points_clean"/>

    <!-- human tracker params -->
    <param name="estimate_frame" type="string" value="position_lidar"/>
    <param name="baseline_estimate" type="string" value="twr_position"/>
    <param name="publish_clusters" type="bool" value="true"/>
    <param name="cluster_topic" type="string" value="points_clusters"/>
    <param name="cluster_min_size" type="double" value="10.0"/>
    <param name="cluster_max_size" type="double" value="200.0"/>
    <param name="human_max_movement" type="double" value="1.2"/>
    <param name="ready_topic" type="string" value="detection_ready"/>

    <node name="$(anon lidar_filter)" pkg="lidar_detector" type="lidar_filter" output="screen" required='true'/>
    <node name="$(anon ground_detect)" pkg="lidar_detector" type="detect_ground.py" output="screen" required='true'/>
    <node name="$(anon human_detect)" pkg="lidar_detector" type="detect_leader.py" output="screen" required='true'/>
</group>

</launch>
